src/tests/functional/plugin/shared/src/behavior/ov_infer_request/infer_querystate.cpp

```diff
@@ -0,0 +1,38 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+#include <gtest/gtest.h>
+#include "behavior/ov_infer_request/infer_querystate.hpp"
+
+namespace ov {
+namespace test {
+namespace behavior {
+std::string OVInferRequestQueryStateTest::getTestCaseName(testing::TestParamInfo<InferRequestParams> obj) {
+    std::string targetDevice;
+    ov::AnyMap configuration;
+    std::tie(targetDevice, configuration) = obj.param;
+    std::replace(targetDevice.begin(), targetDevice.end(), ':', '.');
+    std::ostringstream result;
+    result << "querystate_targetDevice=" << targetDevice << "_";
+    if (!configuration.empty()) {
+        using namespace CommonTestUtils;
+        for (auto& configItem : configuration) {
+            result << "configItem=" << configItem.first << "_";
+            configItem.second.print(result);
+        }
+    }
+    return result.str();
+}
+
+TEST_P(OVInferRequestQueryStateTest, QuerystateTest) {
```

Comment:
please double check if any case overlap or integration can be made with below:
https://github.com/openvinotoolkit/openvino/pull/11590/files#diff-e8c107eadd83bad9ed3c25790b3f8682abc7b3b216977e9883119caa207b104c

Reply:
Updated